content/articles/searching-and-sorting-techniques/index.md

```diff
@@ -0,0 +1,231 @@
+step 3: Examine the value in the counter position of the list.
+
+step 4: Examine to see if the value at that place corresponds to the value you're searching for.
+
+step 5: If it matches, the value has been discovered. Send a message to bring the search to a close.
+
+step 6: If not, go return to step 3 and increase the counter by one until there are no more things to search for.
+
+step 7: Send a message if all of the objects have been searched and no match has been found.
+#### 2. Binary Search
+To search a sorted array, divide the search interval in half regularly. Begin by creating an interval that spans the whole array. If the search key's value is less than the item in the interval's midpoint, the interval should be narrowed to the bottom half. Otherwise, limit it to the upper half of the page. Check the value until it is discovered or the interval is empty.
+##### Algorithm
+The procedures for implementing binary search are as follows:
+
+Step 1: Ask the user for the search element.
+
+Step 2: In the sorted list, locate the center element.
+
+Step 3: Compare the element you are looking for to the sorted list's center element.
+
+Step 4 - If both elements match, the program will display the result and the function will be terminated.
+
+Step 5 - If neither of the elements is matched, see if the element you are looking for is smaller or greater than the center element.
+
+Step 6 - If the element you're looking for isn't in the center, repeat steps 2, 3, 4, and 5 for the lower sublist's center element.
+#### 3. Interpolation Search
+Depending on the value of the key being searched, interpolation search may go to various places. If the key's value is closer to the final element, for example, interpolation search is more likely to begin at the end.
+
+ Interpolation search utilizes the formula below to get the place to be searched.
+```bash
+ pos = i + [ (k-ar[i])*(z-i) / (ar[z]-ar[i]) ]
+ ```
+Let's look at the meaning of this formula:
+- ar[]: Array where elements need to be searched
+- k: Element to be searched
+- i: Starting index in ar[]
+- z: Ending index in ar[]
+##### Algorithm
+Step 1: Begin looking for information in the middle of the list.
+
+Step 2: If there is a match, return the item's index and leave.
+
+Step 3: Probe the position if it isn't a match.
+
+Step 4: Using the probing formula, divide the list and discover the new middle.
+
+Step 5: If the data is larger than the center, look for it in a higher sub-list.
+
+Step 6: If the data is less than the middle, look in the lowest sub-list.
+
+Step 7: Continue until you've found a match.
+#### 4. Jump Search
+The Jump Search Approach is a new sorting method that locates a particular component in a sorted list. When compared to a linear search algorithm, the basic principle behind this searching strategy is to search a smaller amount of elements.
+
+In each iteration, this can be accomplished by skipping a fixed number of array elements or leaping ahead a fixed number of steps.

```

Comment:
```suggestion
In each iteration, this can be accomplished by skipping a fixed number of array elements or leaping ahead a fixed number of steps.

```